sphinx/config.py

```diff
@@ -163,6 +163,12 @@ def read(cls, confdir: str, overrides: Dict = None, tags: Tags = None) -> "Confi
             raise ConfigError(__("config directory doesn't contain a conf.py file (%s)") %
                               confdir)
         namespace = eval_config_file(filename, tags)
+
+        # Resolve https://github.com/sphinx-doc/sphinx/issues/10474 where conf.py
+        # explicitly sets language to None, by coercing it to English.
+        if namespace.get("language", ...) is None:
+            namespace["language"] = "en"
```

Comment:
Thank you for your nice patch. But I think we should emit a warning on fallback not to set `language = None`. I changed the default value of `language` in https://github.com/sphinx-doc/sphinx/pull/10067, and now it takes only a string.

Reply:
We can additionally emit a warning, but per the bug report I would strongly argue we should coerce. 

A

Reply:
I just made some fixes for this PR. Could you check this, please?
https://github.com/sphinx-doc/sphinx/compare/5.x...tk0miya:10474_language_fallback?expand=1

Reply:
Sorry I just saw this, will do.

A

Reply:
If we'll keep accepting `None` value, it would be better to revert https://github.com/sphinx-doc/sphinx/pull/10067 partially instead of this. One of my purpose of #10067 is change the type of `language` to `str` from `Optional[str]`. So automatic conversion is not good idea.

Reply:
@tk0miya I cherry-picked the commits from your branch into mine and updated.

This only affects the `conf.py` and `Config.read` -- in the rest of Sphinx (including `Config.__init__`), `config.language` has type `str`.

A

Reply:
Thanks, I'll check your update tomorrow. Good night!